tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php

```diff
@@ -80,81 +80,219 @@ public function test_register() {
 	}
 
-	public function test_get_rest_routes__success() {
+	public function test_sharing_settings__requires_authenticated_admin() {
+		$this->enable_feature( 'dashboardSharing' );
+
+		$admin = self::factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $admin->ID );
+		$this->controller->register();
+
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/modules/data/sharing-settings' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'search-console' => array(
+						'sharedRoles' => array( 'editor', 'subscriber' ),
+						'management'  => 'all_admins',
+					),
+					'analytics'      => array(
+						'sharedRoles' => array( 'editor' ),
+					),
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEquals( 'rest_forbidden', $response->get_data()['code'] );
+	}
+
+	public function test_sharing_settings__create_new_settings() {
 		$this->enable_feature( 'dashboardSharing' );
-		$module_sharing_settings = $this->modules->get_module_sharing_settings();
 
 		$admin_1 = self::factory()->user->create_and_get( array( 'role' => 'administrator' ) );
 		wp_set_current_user( $admin_1->ID );
-		$initial_sharing_settings = array(
-			'search-console'     => array(
-				'sharedRoles' => array( 'editor', 'subscriber' ),
-				'management'  => 'all_admins',
-			),
-			'analytics'          => array(
-				'sharedRoles' => array( 'editor' ),
-				'management'  => 'owner',   // To test that non-owners cannot merge settings for this module.
+		// Setup the site and admin_1 user to make a successful REST request.
+		$this->grant_manage_options_permission();
+		update_option( 'googlesitekit_search-console_settings', array( 'ownerID' => $admin_1->ID ) );
+
+		$this->controller->register();
+
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/modules/data/sharing-settings' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'search-console' => array(
+						'sharedRoles' => array( 'editor', 'subscriber' ),
+					),
+					'analytics'      => array(
+						'sharedRoles' => array( 'editor' ),
+						'management'  => 'owner',
+					),
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+
+		$expected_response = array(
+			'settings'    => array(
+				// admin_1 is the owner and can add/edit search-console settings but
+				// NOT analytics settings.
+				'search-console'     => array(
+					'sharedRoles' => array( 'editor', 'subscriber' ),
+					'management'  => 'owner',
+				),
+				// Default settings always saved for shared ownership modules.
+				'pagespeed-insights' => array(
+					'sharedRoles' => array(),
+					'management'  => 'all_admins',
+				),
 			),
-			'pagespeed-insights' => array(
-				'sharedRoles' => array( 'editor', 'subscriber' ),
-				'management'  => 'all_admins',
+			'newOwnerIDs' => (object) array(
```

Comment:
Why is the `object` cast needed here? Just seems a bit odd to me.

```suggestion
			'newOwnerIDs' => array(
```

Reply:
Odd indeed! But it is to [maintain the shape of the response](https://github.com/google/site-kit-wp/issues/4481#issuecomment-1130393863) when translated to JSON.